### PR TITLE
Enable HTTP/2 (h2c) with PHP FPM

### PR DESCRIPTION
**Pull request by plyhun**
Built over a sister php:8.1-ppm-bullseye image with extra apache2 installed, to reuse the Apache-based routing configuration. Otherwise all the configs would have a need being transformed into FPM's native format.

### CODE REVIEW

File: portal-files/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,31 @@
 FROM php:8.1-apache-bullseye
 
-# Install Debian packagess
+RUN mkdir -p /run/php/
```
Comment (plyhun) [inline portal-files/Dockerfile]: Required for FPM's PID tracking. If UNIX socket connection is used, the socket is also created here.

File: portal-files/Dockerfile [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+
+# Install Debian packages
 RUN apt-get update -y \
-	&& apt-get install -y \
+    && apt-get install -y software-properties-common ca-certificates lsb-release apt-transport-https wget gnupg \
+    && sh -c 'echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list' \
```
Comment (plyhun) [inline portal-files/Dockerfile]: This is an "official" PHP FPM repo.

File: portal-files/apache2/vhost.conf [hunk 3]
```diff
@@ -8,4 +8,13 @@
 
 	ErrorLog ${APACHE_LOG_DIR}/error.log
 	CustomLog ${APACHE_LOG_DIR}/access.log combined
+	Protocols h2c http/1.1
+	H2Direct on
```
Comment (plyhun) [inline portal-files/apache2/vhost.conf:12]: Recommended setup.

File: portal-files/Dockerfile [hunk 4]
```diff
@@ -1,19 +1,22 @@
...
 RUN apt-get update -y \
-	&& apt-get install -y \
+    && apt-get install -y \
 		zip \
 		unzip \
 		git \
 		pkg-config \
 		zlib1g-dev \
 		jq \
+		libcurl4-openssl-dev \
```
Comment (plyhun) [inline portal-files/Dockerfile:12]: CURL is required for HTTP/2 client calls from Guzzle.

File: portal-files/supervisord-portal-php.conf [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+process_name=apache2
+autostart=true
+autorestart=true
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+stderr_logfile=/dev/stderr
+stderr_logfile_maxbytes=0
+
+[eventlistener:portal-php-healthcheck]
+command=bash -c "printf 'READY\n' && while read line; do kill -SIGQUIT $PPID; done < /dev/stdin"
```
Comment (plyhun) [inline portal-files/supervisord-portal-php.conf:27]: The supervisor is tuned to stop the whole container if any of the underlying processes stops.

File: composer-setup.php
Comment (netwarex) [inline composer-setup.php:1]: Could you please remove this file?

**Review by netwarex — APPROVED** [review]

**Pull request merged**